### PR TITLE
Build: use different versions of jsdom for Node and iojs testing

### PR DESCRIPTION
**Pull request by timmywil**
Fixes gh-2266

### CODE REVIEW

File: build/tasks/install_jsdom.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+		var done = this.async(),
+			pkg = grunt.config( "pkg" ),
+			version = pkg.jsdomVersions[
+
+				// Unfortunately, this is currently the only
+				// way to tell the difference between Node and iojs
+				/^v0/.test( process.version ) ? "node" : "iojs"
+			];
+
+		// Link npm so we can use the global one
```
Comment (mzgol) [inline build/tasks/install_jsdom.js]: Why is that important? Can't we follow the usual pattern & use a local `npm`?

File: build/tasks/install_jsdom.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+		grunt.util.spawn( {
+			cmd: "npm",
+			args: [ "link", "npm" ]
+		}, function() {
+			var npm = require( "npm" );
+			npm.load(function() {
+				npm.commands.ls( [ "jsdom" ], function( data ) {
```
Comment (mzgol) [inline build/tasks/install_jsdom.js]: How fast is it? `npm ls` is slow enough that I built [check-dependencies](http://npmjs.com/package/check-dependencies) to have a quick check for installed deps. Manually doing `require('./node_modules/jsdom/package.json').version` (+ checking if the file exists first) would be faster, less code & I doubt it'll ever stop working.

File: build/tasks/install_jsdom.js (hunk 1 above)
Comment (timmywil) [inline build/tasks/install_jsdom.js]: There is no local npm. We don't include it in dependencies anymore.

File: build/tasks/install_jsdom.js (hunk 2 above)
Comment (timmywil) [inline build/tasks/install_jsdom.js]: Good idea. I'll do that instead.

File: build/tasks/install_jsdom.js (hunk 1 above)
Comment (timmywil) [inline build/tasks/install_jsdom.js]: We could use command-line npm with another spawn. Now that it's just one npm command, I don't mind that as much.

Comment (timmywil) [inline build/tasks/install_jsdom.js]: Changed to use spawn.

File: build/tasks/install_jsdom.js (hunk 2 above)
Comment (timmywil) [inline build/tasks/install_jsdom.js]: Updated.

File: build/tasks/install_jsdom.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+			current = require( "jsdom/package.json" ).version;
+			if ( current === version ) {
+				return;
+			}
+		} catch ( e ) {}
+
+		// Use npm on the command-line
+		// There is no local npm
+		grunt.util.spawn( {
+			cmd: "npm",
```
Comment (mzgol) [inline build/tasks/install_jsdom.js]: Is the indentation OK here?

Comment (timmywil) [inline build/tasks/install_jsdom.js]: Seems to be.

**Comment by mzgol** [comment]
LGTM

**Comment by timmywil** [comment]
Thanks @mzgol. I'm going to go ahead and merge this so our builds start passing.

**Pull request merged**